File: 6-zigzag-conversion/6-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        
        if numRows == 1:
            return s
        res = [""]* numRows
        res[0] = s[0]
        
        #print(res,len(res))
        j = 0 
        i = 1
        while (i<len(s)):
            for j in range(1,numRows):
                if i == len(s):
                    break
                #print(i,j)
                res[j] += s[i]
                i+=1
                
            for j in range(numRows-2,-1,-1):
                if i == len(s):
                    break
                #print(i,j)
                res[j] += s[i]
                i+=1
        ans=""
        for f in res:
            ans+=f
        return ans
                
        
            

File: 6-zigzag-conversion/6-zigzag-conversion.cpp
class Solution
{
public:
  string convert(string s, int numRows)
  {
    if (numRows == 1)
      return s;
      
    vector<string> res(numRows, "");
    res[0] = s[0];
      
    int i = 1;
    int j = 1;
      
    while (i < s.size())
    {
      for (j=1; j < numRows; j++)
      {
        if (i == s.size())
          break;
        res[j] += s[i];
        i++;
      }

      for (j = numRows - 2; j > -1; j--)
      {
        if (i == s.size())
          break;
          
        res[j] += s[i];
        i++;
      }
        
        if(i == s.size())
            break;
    }

      string a = "";
    for (string bug : res)
        a += bug;
      
      return a;
  }
};